support the new glimmer post_menu

fokx opened the pull request:
This PR update the code use support the new glimmer post_menu:
https://meta.discourse.org/t/upcoming-post-menu-changes-how-to-prepare-themes-and-plugins/341014

Also included:
* fix icon name deprecation
* fix `this-property-fallback` deprecation
* replace `I18n.t` with `i18n`
* use `service:site-settings` instead of `site-settings:main`
* linting using `pnpm dlx eslint --fix plugins/discourse-journal`

fokx committed:
support the new glimmer post_menu; use `service:site-settings` instead of `site-settings:main`; fix icon name deprecation; fix `this-property-fallback` deprecation

angusmcleod commented on assets/javascripts/discourse/components/journal-comment-button.gjs:
```diff
@@ -0,0 +1,36 @@
+      class="post-action-menu__comment create fade-out"
+      ...attributes
+      @action={{@buttonActions.replyToPost}}
+      @icon={{if this.args.post.reply_to_post_number "reply" "comment"}}
```
This should probably be a property.

fokx replied:
changed:
https://github.com/fokx/discourse-journal/blob/a32ae058b331cea32e0c8e25c5aceaa9475011a6/assets/javascripts/discourse/components/journal-comment-button.gjs#L34

angusmcleod commented on assets/javascripts/discourse/connectors/after-topic-footer-main-buttons/entry-button-wrapper.hbs:
```diff
@@ -1,4 +1,4 @@
-{{#if topic.can_create_entry}}
+{{#if this.topic.can_create_entry}}
```
This will need to be namespaced to `outletArgs`. See further:

https://meta.discourse.org/t/using-plugin-outlet-connectors-from-a-theme-or-plugin/32727#p-139082-using-outlet-arguments-4

fokx replied:
changed to:
```hbs
{{#if @outletArgs.topic.can_create_entry}}
```

angusmcleod commented on assets/javascripts/discourse/connectors/topic-category/journal-tip-container.hbs:
```diff
@@ -1,4 +1,4 @@
-{{#if topic.showJournalTip}}
+{{#if this.topic.showJournalTip}}
```
Likewise. This is not going to work without `outletArgs`.

fokx replied:
changed to:
```hbs
{{#if @outletArgs.topic.showJournalTip}}
```

angusmcleod reviewed (changes requested):
@fokx Thanks for starting this! Much appreciated. A few comments about the code. Did you test this? Maybe if you could write an acceptance test? Or send a video of it working. Thanks!

Regarding linting, these are the commands I'd recommend (run from the plugin directory).
```
pnpm eslint --no-error-on-unmatched-pattern --fix {test,assets,admin/assets}/javascripts
pnpm prettier --write "{test,assets}/**/*.{scss,js,gjs,es6,hbs}"
pnpm ember-template-lint --no-error-on-unmatched-pattern assets/javascripts
```

fokx commented:
I've tested its main functionality. Here is the screen recording.
[journal-plugin-2025-02-13_22.08.43.svtav1.webm](https://github.com/user-attachments/assets/662fcc64-38b9-41da-b7ef-eb543baa068f)
I will resolve your comments and perhaps write an acceptance test tomorrow/in the weekend.

fokx committed:
use @outletArgs.model; resolve the `this-property-fallback` deprecation

fokx committed:
import standard linting configuration from the latest discourse-plugin-skeleton; remove yarn.lock

fokx committed:
lint using `prettier --write "{test,assets}/**/*.{scss,js,gjs,es6,hbs}"`

fokx committed:
add `ember-template-lint` to devDependencies

fokx committed:
FIX: use @outletArgs.topic/category instead of @outletArgs.model

fokx commented:
After some more manual tests, I found a error I introduced: comment of the original post will not be indented. Sorry for that. 

I find the problem comes with the first commit of last week, that I replaced `openCommentCompose`
```js
        openCommentCompose() {
          const opts = {
            action: Composer.REPLY,
            draftKey: this.model.topic.get("draft_key"),
            draftSequence: this.model.topic.get("draft_sequence"),
            post: this.model,
          };
          api.container.lookup("controller:composer").open(opts);
        }
```
with `@action={{@buttonActions.replyToPost}}`, which is wrong.

fokx committed:
FIX: comment button for original post won't create an entry

fokx commented:
Manual test seems OK. Writing acceptance tests seems too hard for me as of now. I will learn how to do that if I get more free time. Sorry.

angusmcleod commented:
Much better, thanks!
